core/templates/components/ck-editor-helpers/ck-editor-4-rte.component.ts

```diff
@@ -275,7 +277,10 @@ export class CkEditor4RteComponent implements AfterViewInit, OnChanges,
         this.contextService.isExplorationLinkedToStory() &&
               AppConstants.VALID_RTE_COMPONENTS_FOR_ANDROID.indexOf(
                 componentDefn.id) === -1);
-      if (!(hideComplexExtensionFlag || notSupportedOnAndroidFlag)) {
+      if (!(
+        hideComplexExtensionFlag || notSupportedOnAndroidFlag ||
+        this.isInvalidForBlogPostEditorRTE(componentDefn)
+      )) {
```

Comment:
Please move these conditions to a separate function.

Reply:
It is difficult to move these conditions in a separate function since these are being initialized for each RTE component definition in this function itself and the complete code block will have to be moved to a separate function which again will become the same thing.